CMakeLists.txt

```diff
@@ -891,15 +1000,24 @@ set(NO_PTHREADS )
 set(NO_PYTHON )
 set(PAGER_ENV "LESS=FRX LV=-c")
 set(DC_SHA1 YesPlease)
-set(NO_GETTEXT )
 set(RUNTIME_PREFIX true)
 set(X ${EXE})
 if(WIN32)
+	set(NO_GETTEXT YesPlease)
 	set(NO_UNIX_SOCKETS YesPlease)
-else()
+	else()
+	set(NO_GETTEXT )
 	set(NO_UNIX_SOCKETS )
 endif()
 
+#CI builds dont want perl or python
+if(DEFINED CI_BUILD)
+	if(CMAKE_C_COMPILER_ID STREQUAL "MSVC" OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
+		set(NO_PERL 1)
+		set(NO_PYTHON YesPlease)
+	endif()
+endif()
```

Comment:
Can't we do this such that it can be configured via the command-line in the CI run instead? I don't think that _anything_ in `CMakeLists.txt` should be explicitly dependent on whether it runs in a CI or not, that's [too reminiscent of the Diesel scandal](https://www.npmjs.com/package/volkswagen) for my taste :-)

Instead, whatever settings we need to be specified in the CI run should have a fallback default in `CMakeList.txt`, but respect what is being specified via the command-line.

Reply:
We can do this.